src/Credentials/InstanceProfileProvider.php

```diff
@@ -66,6 +78,17 @@ public function __construct(array $config = [])
         $this->retries = (int) getenv(self::ENV_RETRIES) ?: ($config['retries'] ?? self::DEFAULT_RETRIES);
         $this->client = $config['client'] ?? \Aws\default_http_handler();
         $this->ec2MetadataV1Disabled = $config[self::CFG_EC2_METADATA_V1_DISABLED] ?? null;
+        $this->endpoint = Validator::validateURI(
```

Comment:
what do you think about something like:

```
 $this->endpoint = isset($config['endpoint']) && $this->isValidEndpoint($config['endpoint']) 
    ? $config['endpoint'] 
    : null;
```

`endpoint` is the key we should be looking for in config